src/System.Private.CoreLib/shared/System/Threading/ExecutionContext.cs

```diff
@@ -305,20 +277,11 @@ internal static void RunFromThreadPoolDispatchLoop(Thread threadPoolThread, Exec
             CheckThreadPoolAndContextsAreDefault();
             // ThreadPool starts on Default Context so we don't need to save the "previous" state as we know it is Default (null)
 
-            if (executionContext != null && executionContext.m_isDefault)
-            {
-                // Default is a null ExecutionContext internally
-                executionContext = null;
```

Comment:
In this method this value is never used again so no point in setting it to `null`